source/iface/usm_allocations.rst

```diff
@@ -304,6 +304,104 @@ passed to ``sycl::free`` to avoid a memory leak.
 
 See :ref:`usm-implicit-data-movement` for usage.
 
+
+==================================
+Parameterized allocation functions
+==================================
+
+The functions bellow take a ``kind`` parameter that specifies the type of
+USM to allocate. When ``kind`` is ``usm::alloc::device``, then the
+allocation device must have ``aspect::usm_device_allocations``.
+When kind is ``usm::alloc::host``, at least one device in the allocation
+context must have ``aspect::usm_host_allocations``. When ``kind`` is
+``usm::alloc::shared``, the allocation device must have
+``aspect::usm_shared_allocations``. If these requirements are violated,
+the allocation function throws a synchronous ``exception`` with the
+``errc::feature_not_supported`` error code.
+
+================
+``sycl::malloc``
+================
+
+::
+
+   void* sycl::malloc(size_t numBytes, const device& syclDevice,
+                   const context& syclContext, usm::alloc kind,
+                   const property_list& propList = {});
+
+   template <typename T>
+   T* sycl::malloc(size_t count, const device& syclDevice,
+                   const context& syclContext, usm::alloc kind,
+                   const property_list& propList = {});
+
+   void* sycl::malloc(size_t numBytes, const queue& syclQueue, usm::alloc kind,
+                   const property_list& propList = {});
+
+   template <typename T>
+   T* sycl::malloc(size_t count, const queue& syclQueue, usm::alloc kind,
+                   const property_list& propList = {});
+
+   void* sycl::aligned_alloc(size_t alignment, size_t numBytes,
+                    const device& syclDevice, const context& syclContext,
+                    usm::alloc kind, const property_list& propList = {});
+
+   template <typename T>
+   T* sycl::aligned_alloc(size_t alignment, size_t count, const device& syclDevice,
+                    const context& syclContext, usm::alloc kind,
+                    const property_list& propList = {});
+
+   void* sycl::aligned_alloc(size_t alignment, size_t numBytes,
+                    const queue& syclQueue, usm::alloc kind,
+                    const property_list& propList = {});
+
+   template <typename T>
+   T* sycl::aligned_alloc(size_t alignment, size_t count, const queue& syclQueue,
+                    usm::alloc kind, const property_list& propList = {});
+
+
+On successful USM shared allocation, these functions return a pointer to the
```

Comment:
@mkinsner do we need to add missing 'When the allocation size is zero bytes (numBytes or count is zero)' part also here?

Reply:
Yes.  Thanks for pointing this out.  That wording should be in this section as well.

Reply:
I did the same in the PR against the SYCL spec through https://github.com/KhronosGroup/SYCL-Docs/pull/485/commits/49dee708c753766f81701ee00dc94f65b2d132f5

Reply:
Fixed with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/87/commits/7f4abab20a2024f70ee195d78e9b520885123cba

@mkinsner please recheck the paragraph above in spec "On success, these functions return a pointer to the newly allocated memory"...looks like this part should be reworked in order to keep consistency

Reply:
> @mkinsner please recheck the paragraph above in spec "On success, these functions return a pointer to the newly allocated memory"...looks like this part should be reworked in order to keep consistency

I don't quite understand what you're suggesting.  Do you mean that we can clarify the SYCL spec by saying "On successful USM allocation" instead of "On success"?  Or am I misunderstanding?

Reply:
Sorry, I overlooked more than one difference counted in context...yes definitely result is the same and no sense for rework.

Reply:
Sounds good.